### ISSUE TITLE
Has anyone had success with an M720q yet?

### ISSUE THREAD

**Issue by tjes** [badger707/m920q-pcie-bifurcation#2]
Has anyone had any success with making this work on an M720q yet? I followed the guide but reversing the PCIe lane ordering seems to make any device in the PCIe slot undetectable. 

If you remove R885 (disabling lane reversal) and only populate 1 of the NVME drives it detects it. I'm using the same device as frankol in #1 to test.

I had an M920q that I was able to test with too and that had no issues. That makes me think I'm missing something on the M720q but I'm not sure where to begin looking.

M920q with x2 Crucial P5 1TB SSDs in a PH45 and x1 Hynix 256gb SSD in M.2 slot 1.
![image](https://github.com/badger707/m920q-pcie-bifurcation/assets/5140687/7bd3abfa-2cde-4928-bb4e-43e855cfcaae)

M720q with same configuration, only the onboard M.2 in slot 1 is detected
![image](https://github.com/badger707/m920q-pcie-bifurcation/assets/5140687/4820bcda-2bcf-4526-9486-eb51cfd8ab8b)

**Comment by dxx-223** [comment]
[Edit: added "conclusive text" from Intel Ark plus minor tidy-up]

I've been down this path and I think the key difference is that the M720q has a B360 PCH and the M920q has a Q370 chipset.

According to the [Intel Ark page for the B360](https://ark.intel.com/content/www/us/en/ark/products/133332/intel-b360-chipset.html) the "Supported **Processor** PCI Express Port Configuration[s]" is only "**1x16**".
Whereas the [Intel Ark page for the Q370](https://ark.intel.com/content/www/us/en/ark/products/133282/intel-q370-chipset.html) shows "1x16 or 2x8 or **1x8+2x4**".

If you click on the "?" to the right of "Supported Processor PCI Express Port Configurations" there's this pretty conclusive text:

**Note: The processor's actual PCI express configurations will be determined or limited by the value of this chipset attribute even if the processor is capable of additional configurations.**

This makes me think that CFG[6:5] are ignored or unsupported due to the B360.
Adding a resistor to pull CFG[2] to ground (i.e. value "0") does reverse the lane ordering.

But remember that even if you use a x16 Lenovo riser:
_the motherboard only routes the first 8 PCIe lanes from the CPU to the socket_.

![m720q and m920q CPU to PCI x8](https://github.com/badger707/m920q-pcie-bifurcation/assets/68270220/8a55cef7-9f94-4a49-a402-c588df4fde0b)

So by reversing the lane ordering, and being stuck in 1x16 config, CPU lanes marked 0 - 7 are now lanes 15 - 8, and nothing works because there are no lanes 0 - 7.

**TL;DR: bifurcation doesn't seem to be supported on the M720q due to B360 PCH limiting support to 1x16.**
Sell the M720q and buy an M920q instead. Don't forget to remove R885 first!

**Comment by tjes** [comment]
More or less what I ended up figuring out too after reading through the schematics. Figures I missed the part on Intel Ark - that would have saved me a lot of time. :)

**Comment by dxx-223** [comment]
Based on #1 I'm not so confident about my above claim.
I was testing using an M.2 USB card, not an M.2 NVMe and that muddied my results somewhat.

The PCIe lanes are direct from CPU to the PCIe slot, so how would the B360 PCH affect them?
The B360 PCH _does_ supply an additional 4 lanes to the slot, I'm guessing for DMI in case there's a GPU in the slot, but I think there is separate configuration for PCH PCI express lanes.

Maybe it is still possible to enable bifurcation on an M720q.
Could it be that your CFG6 / R889 resistor wasn't working?
If you only had lane-reversal (CFG2 / R885) in effect but not CFG6 / R889 then the slot would have had zero lanes.

I'd re-test on my own M720q but my eyes are still recovering from last time!

**Comment by gotrekg** [comment]
Hi,
I know this is closed but i would like to share mine result of adding more m.2 nvme ssd into m720Q using this board https://github.com/qq8322302/LenovoTiny5PcieCard  no need to do hardware modding just BIOS mod and you will need 8x card to "active" all the m.2 slots

**Comment by SNMetamorph** [comment]
> Hi, I know this is closed but i would like to share mine result of adding more m.2 nvme ssd into m720Q using this board https://github.com/qq8322302/LenovoTiny5PcieCard no need to do hardware modding just BIOS mod and you will need 8x card to "active" all the m.2 slots

This is a good catch.
